gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2568,8 +2563,8 @@ public static String GET_SNAPSHOT_TXID() throws SQLException, StandardException
     }
     // tc commit will clear all the artifacts but will not commit actual txState
     // that should be committed in COMMIT procedure
-    tc.resetActiveTXState(true);
-    TXManagerImpl.getOrCreateTXContext().clearTXState();
+    tc.clearActiveTXState(true, true);
+    //tc.resetActiveTXState(true);
```

Comment:
The difference between the two is the reset will change to NOT_SET so it will read from thread-local the next time. Whereas clear will make it null both in GemFireTransaction and in thread-local meaning the transaction is gone. I think need to think on what this means here.

Why is existing transaction being cleared in GET in the first place? The returned txId is no longer valid, so is the caller supposed to start a transaction with the same txId later? Can you write a short document on the flow of GET, COMMIT, ROLLBACK calls and how the caller is supposed to use those?